src/utils/diagnostics.jl

```diff
@@ -0,0 +1,18 @@
+function find_type_instabilities(m, comp, tool=InteractiveUtils.code_warntype)
+    m.mi===nothing && error("Model must be built.")
+    
+    mi = m.mi
+
+    time_keys::Vector{Int} = dim_keys(mi.md, :time)
+    clock = Clock(time_keys)
+
+    dim_dict_keys = collect(keys(getproperty(mi.comps_dict[comp].comp_id.module_obj, mi.comps_dict[comp].comp_id.comp_name).dim_dict))
+    dim_dict = OrderedDict([k => m.md.dim_dict[k] for k in dim_dict_keys])
+
+    f = mi.comps_dict[comp].run_timestep
+    v = typeof(mi.comps_dict[comp].variables)
+    p = typeof(mi.comps_dict[comp].parameters)    
+    d = typeof(NamedTuple(name => (name == :time ? timesteps(clock) : collect(values(dim))) for (name, dim) in dim_dict))
+
+    tool(f, (p, v, d, Int))
```

Comment:
@davidanthoff why does putting `Int` here work? I was hoping to make very similar function that returned just `run_timestep` so I could run it with @profileview` ... but seem to get errors when I try to just run:

```
f(p,v,d,Int)
```

am I missing something obvious here? Is the `Int` useful because of the specific `code_warntype` tool ... and I need to be fancier to just return the function?

Reply:
I think we want to put a `AbstractTimestep` type in there to get much more accurate results, for example
```
Mimi.FixedTimestep{1750, 1, 2300}
```
we should be able to get that from the `clock` or something, that is parameterized by the Timestep type I think, I'll take a look tomorrow.